delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/CompositeViews.scala

```diff
@@ -128,6 +494,109 @@ object CompositeViews {
+      orgs: Organizations,
+      projects: Projects,
+      acls: Acls
+  )(implicit
+      uuidF: UUIDF,
+      clock: Clock[UIO],
+      as: ActorSystem[Nothing],
+      baseUri: BaseUri,
+      rmr: RemoteContextResolution
+  ): Task[CompositeViews] = {
+
+    def validateAcls(cpSource: CrossProjectSource) = {
+      val aclAddress = AclAddress.Project(cpSource.project)
+      acls.fetchWithAncestors(aclAddress).map(_.exists(cpSource.identities, resources.read, aclAddress)).flatMap {
+        case true  => IO.unit
+        case false => IO.raiseError(CrossProjectSourceForbidden(cpSource))
+      }
+    }
+
+    def validateProject(cpSource: CrossProjectSource) = {
+      projects.fetch(cpSource.project).leftMap(_ => CrossProjectSourceProjectNotFound(cpSource))
+    }
+
+    def validateSource: ValidateSource = {
+      case _: ProjectSource             => IO.unit
+      case cpSource: CrossProjectSource =>
+        for {
+          _ <- validateAcls(cpSource)
+          _ <- validateProject(cpSource)
+        } yield ()
+      //TODO add proper validation when we'll have delta client for indexing
+      case _: RemoteProjectSource       => IO.unit
+    }
+
+    def validatePermission(permission: Permission) =
+      permissions.fetchPermissionSet.flatMap { perms =>
+        IO.when(!perms.contains(permission))(IO.raiseError(PermissionIsNotDefined(permission)))
+      }
+
+    def validateProjection: ValidateProjection = {
+      case sparql: SparqlProjection    => validatePermission(sparql.permission)
+      //TODO add validation of Es index
+      case es: ElasticSearchProjection => validatePermission(es.permission)
+    }
+
+    apply(config, eventLog, orgs, projects, validateSource, validateProjection)
+  }
+
+  private[compositeviews] def apply(
+      config: CompositeViewsConfig,
+      eventLog: EventLog[Envelope[CompositeViewEvent]],
+      orgs: Organizations,
+      projects: Projects,
+      validateSource: ValidateSource,
+      validateProjection: ValidateProjection
+  )(implicit
+      uuidF: UUIDF,
+      clock: Clock[UIO],
+      as: ActorSystem[Nothing],
+      rmr: RemoteContextResolution
+  ): Task[CompositeViews] = for {
+    agg          <- aggregate(config, validateSource, validateProjection)
+    index        <- UIO.delay(cache(config))
+    sourceDecoder = CompositeViewFieldsJsonLdSourceDecoder(uuidF)
+    views         = new CompositeViews(agg, eventLog, index, orgs, projects, sourceDecoder)
+
+  } yield views
+
+  private def aggregate(config: CompositeViewsConfig, validateS: ValidateSource, validateP: ValidateProjection)(implicit
+      as: ActorSystem[Nothing],
+      uuidF: UUIDF,
+      clock: Clock[UIO]
+  ) = {
+
+    val definition = PersistentEventDefinition(
+      entityType = moduleType,
+      initialState = Initial,
+      next = next,
+      evaluate = evaluate(validateS, validateP, config.maxSources, config.maxProjections),
+      tagger = EventTags.forProjectScopedEvent(moduleType),
```

Comment:
You need to add also add a tag for the generic "view". See ElasticSearchViews or BlazegraphViews for the example